bin/pre-commit.sh

```diff
@@ -1,46 +1,25 @@
 #!/bin/bash
-# Runs pre-commit tests.
-#
-# Instead of failing on first error, complete all checks, then fail if need be.
 
-# Echo go's version to verify against go's behavior below.
-# Different versions do different things with respect to vendored code.
-go version
+rc=0
 
-# Assert state.
-if [ -n "$failIt" ]; then
-  echo "Expecting failIt to be empty."
-  exit 1
-fi
+go_dirs() {
+  go list -f '{{.Dir}}' ./... | tr '\n' '\0'
+}
 
-wantEmpty=$(gofmt -s -d -l . 2>&1 )
-if [ -n "$wantEmpty" ]; then
-  printf >&2 '\ngofmt failed for:\n%s\n' "$wantEmpty"
-  failIt=1
-fi
+echo "Running go fmt"
+go_dirs | xargs -0 gofmt -s -d -l
```

Comment:
Observation: If any subdirectory has a `vendor` directory then this would also run `gofmt` on that - and will likely fail. I think that is fair, as we want to push for repo-global `vendor` anyway.

Reply:
Let's merge it that way, so that we can move forward, I'll try to come-up with something better :D